src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1217,7 +1217,12 @@ void Partitioner::saveRepeatedConstants(const std::string& func_name) {
 
     // Now walk through through every Const bank and inspect the above properties
     auto values_are_the_same = [](const CTPtr& node_a, const CTPtr& node_b) {
-        switch (node_a->output(0).get_element_type()) {
+        const auto node_a_out = node_a->output(0);
+        // NB: Only consider first 4bits in case single value constant (e.g zp) and u4 element type
+        if (node_a_out.get_element_type() == ov::element::u4 && ov::shape_size(node_a_out.get_shape()) == 1u) {
+            return (node_a->get_vector<uint8_t>()[0] & 0x0f) == (node_b->get_vector<uint8_t>()[0] & 0x0f);
+        }
```

Comment:
The proper fix should go to get_vector<T> to avoid garbage is the 4 hi bits if the vector size is odd.

If you want to fix it here, please:
1. Handle `i4` too
2. Handle the odd vector size.

This case should probably go to a dedicated function.

Reply:
Based on @praasz suggestion this is expected behavior for `get_vector`. To properly handle such cases we need to use `cast_vector<uint8_t>(1)` instead.

Reply:
Looks it's already handled there (see: https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/constant.hpp#L333-L347)

@dmatveev What do you think?

Reply:
@TolyaTalamanov 
#26177
Check if can help.

Reply:
> Looks it's already handled there (see: https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/constant.hpp#L333-L347)
> 
> @dmatveev What do you think?

@TolyaTalamanov does the above works for you? If not, let the folks know as there's time to fix.

Reply:
Problem still exists